### PR TITLE
Refactor/api/tokens

### PR DESCRIPTION
**Pull request by Velenir**
Changes to `api/Tokens.ts` and related types to bring them up to date with the contracts

Major changes -- two new functions:
* depositETH
* withdrawETH

only available for `TokenETH`, that's why they aren't generic like all others from `ERC20Interface`

Do we need any other `Token*` methods exposed like that in FE?

### CODE REVIEW

File: src/api/contracts.ts [hunk 1]
```diff
@@ -9,14 +16,18 @@ const contractNames = [
   'Token',
   'TokenETH',
```
Comment (W3stside) [inline src/api/contracts.ts]: should be `Ether Token` here

Comment (W3stside) [reply src/api/contracts.ts]: applicable everywhere else as well below

File: src/api/index.ts [hunk 2]
```diff
@@ -21,10 +21,11 @@ export const getAllAccounts = async () => {
 }
 
 // ether balance, not ETH tokens
-export const getETHBalance = async () => {
+export const getETHBalance = async (account?: Account) => {
```
Comment (W3stside) [inline src/api/index.ts]: `getETHBalance` doesn't exist for me on the `web3` object. Is it not `web3.eth.getBalance(...)`?

File: src/api/types.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export type Index = number | BigNumber
 export interface ProviderInterface {
   getCurrentAccount(): Promise<Account>,
   getAccounts(): Promise<Account[]>,
-  getETHBalance(account?: Account): Promise<BigNumber>,
+  getETHBalance(account: Account): Promise<BigNumber>,
```
Comment (W3stside) [inline src/api/types.ts]: ignore if previous comment is wrong but if not these need to change as well

File: src/api/types.ts [hunk 4]
```diff
@@ -62,21 +65,82 @@ export type Filter = 'latest' | 'pending' | FilterObject | void
...
 export interface ERC20Interface {
   address: Account,
-  getTotalSupply(): Promise<BigNumber>,
-  balanceOf(account?: Account): Promise<BigNumber>,
-  transfer(to: Account, value: Balance, sender: Account, tx?: TransactionObject): Promise<Receipt>,
-  transferFrom(sender: Account, to: Account, value: Balance, tx?: TransactionObject): Promise<Receipt>,
-  approve(spender: Account, value: Balance, sender: Account, tx?: TransactionObject): Promise<Receipt>,
+  totalSupply(): Promise<BigNumber>,
+  balanceOf(account: Account): Promise<BigNumber>,
+  transfer(to: Account, value: Balance, tx?: TransactionObject): Promise<Receipt>,
```
Comment (W3stside) [inline src/api/types.ts]: IMO - as these only return a `bool` based on transfer success or not, this, `transferFrom && approve` should have a manadatory `tx` param

**Review by W3stside — CHANGES_REQUESTED** [review]

File: src/api/types.ts (hunk 4 above)
Comment (Velenir) [reply src/api/types.ts]: Yes, you're right

**Review by W3stside — APPROVED** [review]
Nice stuff, pls merge

**Pull request merged**